main.go

```diff
@@ -133,20 +154,62 @@ func (p *Parser) CountLOC(filePath string) (int, int) {
 		}
 	}
 
-	return loc, cloc
+	return loc, cloc, assertions, tests
 }
 
 type TextReport struct {
+}
 
+//JsonReport json structure for LOC report
+type JsonReport struct {
+	LOC struct {
+		CLOC  int
+		NCLOC int
+	}
+	Imports    int
+	Structs    int
+	Interfaces int
+	Methods    struct {
+		Total    int
+		Exported int
+	}
+	Functions struct {
+		Total    int
+		Exported int
+	}
+	Testing struct {
+		Cases      int
+		Assertions int
+	}
 }
 
-func (t *TextReport) Print(res *Result) {
-	fmt.Printf("LOC:        %v (%v CLOC, %v NCLOC)\n", res.LOC, res.CLOC, res.NCLOC)
-	fmt.Printf("Imports:    %v\n", res.Import)
-	fmt.Printf("Structs:    %v\n", res.Struct)
-	fmt.Printf("Interfaces: %v\n", res.Interface)
-	fmt.Printf("Methods:    %v (%v Exported)\n", res.Method, res.ExportedMethod)
-	fmt.Printf("Functions:  %v (%v Exported)\n", res.Function, res.ExportedFunction)
+func (t *TextReport) Print(res *Result, outputFmt string) {
```

Comment:
This is slightly wrong. There should be an interface called ReportInterface which just has the print method. Then each report format (TextReport, JsonReport, etc.) can be implemented as a struct implementing that interface. You still need a switch in the main() function to create the required report format.